airflow/api_connexion/endpoints/user_endpoint.py

```diff
@@ -62,3 +64,43 @@ def get_users(limit, order_by='id', offset=None):
     users = query.offset(offset).limit(limit).all()
 
     return user_collection_schema.dump(UserCollection(users=users, total_entries=total_entries))
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_USER)])
+def post_user():
+    """Create a new user"""
```

Comment:
We can add a check that AUTH_USER_REGISTRATION is True and return early if disabled?

Reply:
Oh, yes we should, will do. (What error should we return, 405 Method Not Allowed?)

Reply:
405 doesn't quite feel right from an HTTP/REST stand point.

422 might work (even if it's strictly from WebDAV) https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

Reply:
BadRequest will be fine as I think that it's a user error

Reply:
> 405 doesn't quite feel right from an HTTP/REST stand point.
> 
> 422 might work (even if it's strictly from WebDAV) https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

Cool

Reply:
Thinking about this again, I wonder whether a 5xx would make sense, since a 4xx indicates the client can fix something to make the request work, but no amount of client-side changes can make this error go away, it is only fixable at the server side (and once it’s resolved, the client can re-send the request unchanged, which is not typical of 4xx either).

Maybe something like 501 Not Implemented?

Reply:
I kind of feel 400 is not appropriate, since the API client does not know whether a server allows registration or not. I’ll use 403 for now since MDN’s description seems to somehow make sense in this context:

>  The access is permanently forbidden and tied to the application logic, such as insufficient rights to a resource.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403